PWGHF/D2H/Tasks/taskLbReduced.cxx

```diff
@@ -0,0 +1,825 @@
+          }
+        } else {
+          registry.add("hMassPtCutVarsRecSig", "#Lambda_{b}^{0} candidates (matched);#it{M} (D#pi) (GeV/#it{c}^{2});#it{p}_{T}(#Lambda_{b}^{0}) (GeV/#it{c});#Lambda_{b}^{0} candidate decay length (cm);#Lambda_{b}^{0} candidate norm. decay length XY (cm);#Lambda_{b}^{0} candidate impact parameter product (cm);#Lambda_{b}^{0} candidate cos(#vartheta_{P});#it{M} (pK#pi) (GeV/#it{c}^{2});#it{p}_{T}(#Lambda_{c}^{#plus}) (GeV/#it{c});#Lambda_{c}^{+} candidate ML score bkg;#Lambda_{c}^{+} candidate ML score nonprompt", {HistType::kTHnSparseF, {axisMassLb, axisPtLb, axisDecayLength, axisNormDecayLength, axisImpParProd, axisCosp, axisMassLc, axisPtLc, axisMlScore, axisMlScore}});
+          if (fillBackground) {
+            registry.add("hMassPtCutVarsRecBg", "#Lambda_{b}^{0} candidates (unmatched);#it{M} (D#pi) (GeV/#it{c}^{2});#it{p}_{T}(#Lambda_{b}^{0}) (GeV/#it{c});#Lambda_{b}^{0} candidate decay length (cm);#Lambda_{b}^{0} candidate norm. decay length XY (cm);#Lambda_{b}^{0} candidate impact parameter product (cm);#Lambda_{b}^{0} candidate cos(#vartheta_{P});#it{M} (pK#pi) (GeV/#it{c}^{2});#it{p}_{T}(#Lambda_{c}^{#plus}) (GeV/#it{c});#Lambda_{c}^{+} candidate ML score bkg;#Lambda_{c}^{+} candidate ML score nonprompt", {HistType::kTHnSparseF, {axisMassLb, axisPtLb, axisDecayLength, axisNormDecayLength, axisImpParProd, axisCosp, axisMassLc, axisPtLc, axisMlScore, axisMlScore}});
+          }
+        }
+      }
+    }
+  }
+
+  /// Selection of Lb daughter in geometrical acceptance
+  /// \param etaProng is the pseudorapidity of Lb prong
+  /// \param ptProng is the pT of Lb prong
+  /// \return true if prong is in geometrical acceptance
+  template <typename T = float>
+  bool isProngInAcceptance(const T& etaProng, const T& ptProng)
+  {
+    return std::abs(etaProng) <= etaTrackMax && ptProng >= ptTrackMin;
+  }
+
+  /// Fill candidate information at reconstruction level
+  /// \param doMc is the flag to enable the filling with MC information
+  /// \param withDecayTypeCheck is the flag to enable MC with decay type check
+  /// \param withLcMl is the flag to enable the filling with ML scores for the Lc daughter
+  /// \param withLbMl is the flag to enable the filling with ML scores for the Lb candidate
+  /// \param candidate is the Lb candidate
+  /// \param candidatesLc is the table with Lc candidates
+  template <bool doMc, bool withDecayTypeCheck, bool withLcMl, bool withLbMl, typename Cand, typename CandsLc>
+  void fillCand(Cand const& candidate,
+                CandsLc const&)
+  {
+    auto ptCandLb = candidate.pt();
+    auto invMassLb = hfHelper.invMassLbToLcPi(candidate);
+    auto candLc = candidate.template prong0Lc_as<CandsLc>();
+    auto ptLc = candidate.ptProng0();
+    auto invMassLc = candLc.invMassHypo0();
```

Comment:
why only `Hypo0` is considered? I think that here both `Hypo0` or `Hypo1` should be considered, based on the selection flag of the `Lc`

Reply:
fixed fellow with the Bs for the moment